### PR TITLE
Update CLI after bumping version to 2.3.13

### PR DESCRIPTION
**Pull request by florianmahner**

### CODE REVIEW

File: thingsvision/thingsvision.py [hunk 1]
```diff
@@ -36,9 +36,9 @@ def get_parsers():
     common_parser.add_argument(
         "--device",
         type=str,
-        default="cpu",
+        default="cuda",
         choices=["cpu", "cuda"],
-        help="Device to use for the extractor. (default: cpu)",
+        help="Device to use for the extractor. (default: cuda)",
```
Comment (Alxmrphi) [inline thingsvision/thingsvision.py:41]: Ah, I was just about to comment on the default and then the next commit came in that fixed it. I was going to mention about adding in MPS options but that can be for a separate PR now, I think.

File: thingsvision/thingsvision.py [hunk 2]
```diff
@@ -66,24 +66,19 @@ def get_parsers():
...
-        type=str,
-        help="optional list of class names for class dataset. (default: None)",
-        default=None,
-    )
-    parser_extract.add_argument(
-        "--file-names",
-        type=str,
-        help="optional list of file names for class dataset. (default: None)",
-        default=None,
-    )
```
Comment (Alxmrphi) [inline thingsvision/thingsvision.py:80]: References to these also need to probably be removed from ImageDataset in  utils/data/dataset.py if we're taking them out of action here. Actually, I'm not 100% sure why they are being removed, specifying the ordering of an `ImageDataset` seems potentially useful.

File: thingsvision/thingsvision.py [hunk 3]
```diff
@@ -145,18 +140,17 @@ def main():
         elif args.command == "extract-features":
             parser_extract.print_help(sys.stderr)
         sys.exit(1)
-
-    device = torch.device(args.device)
+    
+    device = 'cuda' if torch.cuda.is_available() and args.device == "cuda" else 'cpu'
```
[the remark this answers is not in the document]
Comment (LukasMut) [reply thingsvision/thingsvision.py:144]: I think this is a great idea. A warning should fix that issue.

File: thingsvision/thingsvision.py (hunk 2 above)
Comment (andropar) [reply thingsvision/thingsvision.py:80]: I guess the thinking here is that the CLI is the most basic way to get to CNN features from a directory of images (which will create a list of the image order anyway), so if someone wants to specify the order, they should probably just do programatically anyways? In that sense, we also would not need to remove it from the ImageDataset.

File: thingsvision/thingsvision.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (andropar) [reply thingsvision/thingsvision.py:144]: I agree that we should avoid unintended behaviour. Does `torch.device()` raise an error when the device does not exist? Then we could just leave it like that.

[the remark this answers is not in the document]
Comment (andropar) [reply thingsvision/thingsvision.py:144]: OTOH, if the default is `cuda` this might be a bit confusing. I guess giving a warning would be the best option then!

[the remark this answers is not in the document]
Comment (florianmahner) [reply thingsvision/thingsvision.py]: Have added a check of the device into the `BaseExtractor` that also triggers a warning if `cuda` is not available. Generally,I think its much cleaner to have all checks and exception handling outside of the CLI.

[the remark this answers is not in the document]
Comment (LukasMut) [reply thingsvision/thingsvision.py]: I agree with @florianmahner that we should have all such checks outside of the CLI but these checks are definitely important to add.

File: thingsvision/thingsvision.py (hunk 2 above)
Comment (LukasMut) [reply thingsvision/thingsvision.py:80]: Having the possibility to add optional class and file names is pretty useful and I would not remove it from the dataset class. OTOH it's quite cumbersome to append all class or file names in your shell to the respective argument for using the CLI. No one whose intention is to perform quick and basic extraction would do that. At least I wouldn't.

**Review by LukasMut — APPROVED** [review]

File: thingsvision/thingsvision.py (hunk 1 above)
Comment (LukasMut) [reply thingsvision/thingsvision.py:41]: @Alxmrphi, would you be willing to submit that PR after I've accepted the current one?

**Comment by LukasMut** [comment]
> Mainly comments on ideas from my end, for this. However, the only change that needs to happen to keep things consistent is an update of `dataset.py` in the `utils` folder if we're no longer going with using class or file name options.

An update of `dataset.py` in the `utils` does not need to happen. See @andropar and my comments in the review.

File: thingsvision/thingsvision.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (florianmahner) [reply thingsvision/thingsvision.py]: I have noticed that we should also check for multiple `cuda:x` device types and that the user should be able to directly select a GPU. Have changed this in `BaseExtractor` and the `CLI` accordingly, I think we should be ready to go now.

[the remark this answers is not in the document]
Comment (LukasMut) [reply thingsvision/thingsvision.py]: Sounds good. Thanks @florianmahner!

**Comment by LukasMut** [comment]
@Alxmrphi and @andropar if the changes made by @florianmahner satisfy your requests, then please approve the PR, else re-request changes. We always need 2 approvals.

**Review by Alxmrphi — APPROVED** [review]
Great discussions regarding some of the potential issues. Questions all resolved.
